### ISSUE TITLE
rpc not working as expected on beacon node

### ISSUE THREAD

**Issue by mkinney** [prysmaticlabs/prysm#3817]
I was experimenting with https://prylabs.net/participate and was able to get it working without any issues. Good job!

I wanted to see if the rpc was working. (just experimenting)

I added ' -p 7000:7000' and '--grpc-gateway-port=7000' to the beacon-chain:latest docker command.

I see the port 7000 is open but I'm getting:

`curl: (56) Recv failure: Connection reset by peer`

when I try:

`curl -vvv http://localhost:7000/eth/v1alpha1/node/version`

Note: I do see:
`INFO gateway: Starting gRPC gateway. address=127.0.0.1:7000`

I may have the endpoint address incorrectly. It may be nice to show the full URI or something.

**Comment by mkinney** [comment]
Thanks for making the code changes/PR but I still think there is an issue.

I see the docker images changed.

I removed my docker images and re-pulled the latest.

I started it like this:

```
docker run -it -v /tmp/prysm-data:/data -p 4000:4000 -p 7000:7000 \
>   gcr.io/prysmaticlabs/prysm/beacon-chain:latest \
>   --datadir=/data \
>   --no-genesis-delay \
> --init-sync-no-verify --grpc-gateway-port=7000
```

Here's the top of the log (showing the commit version):

```
[2019-10-22 01:19:09]  INFO node: Starting beacon node version=Git commit: a440c32155c8491cb976498f009af4d1b892ac05. Built at: 2019-10-22 00:46:04+00:00
[2019-10-22 01:19:09]  INFO registry: Starting 9 services: [*p2p.Service *powchain.Service *operations.Service *blockchain.Service *initialsync.InitialSync *sync.RegularSync *rpc.Service *gateway.Gateway *prometheus.Service]
[2019-10-22 01:19:09]  INFO powchain: Connected to eth1 proof-of-work chain endpoint=wss://goerli.prylabs.net/websocket
[2019-10-22 01:19:09]  INFO rpc: RPC-API listening on port port=:4000
[2019-10-22 01:19:09]  WARN rpc: You are using an insecure gRPC connection! Provide a certificate and key to connect securely
[2019-10-22 01:19:09]  INFO gateway: Starting gRPC gateway. address=127.0.0.1:7000
```

Port is open:

```# netstat -antd | grep 7000
tcp6       0      0 :::7000                 :::*                    LISTEN
```

Still get same error:

```
# curl -X GET "http://localhost:7000/eth/v1alpha1/node/version" -H "accept: application/json"
curl: (56) Recv failure: Connection reset by peer
```

Any ideas?

**Comment by mkinney** [comment]
I uninstalled docker.io and installed docker-ce by using https://docs.docker.com/install/linux/docker-ce/ubuntu/

Same results.

Trying from docker on mac as well. I get a slightly different error:

```
$ curl "http://127.0.0.1:7000/eth/v1alpha1/node/version"
curl: (52) Empty reply from server
```

**Comment by mkinney** [comment]
I built locally on mac using bazel:

```
$ curl http://localhost:7000/eth/v1alpha1/node/version
{"version":"Git commit: {STABLE_GIT_COMMIT}. Built at: {DATE}"}
```

So, I don't think it's a code issue at this point.

**Comment by rauljordan** [comment]
@mkinney is this resolved by #3868?

**Comment by mkinney** [comment]
With help from @prestonvanloon I was able to fix/test the issue.

Did need to change the localhost to `0.0.0.0`.

For posterity, here is how I was able to test it. (from mac)

`DOCKER_TAG=latest bazel run //beacon-chain:image_bundle --platforms=@io_bazel_rules_go//go/toolchain:linux_amd64_cgo`

`docker run -v $HOME/prysm-data:/data -p 4000:4000 -p 7000:7000   --name beacon-node   gcr.io/prysmaticlabs/prysm/beacon-chain:latest   --datadir=/data   --no-genesis-delay   --grpc-gateway-port=7000`
